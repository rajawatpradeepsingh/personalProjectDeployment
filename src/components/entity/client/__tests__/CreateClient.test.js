import React from "react";
import axios from "axios";
import { render, cleanup, act } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import CreateClient from "../CreateClient";

jest.mock("axios");
afterEach(cleanup);
afterEach(() => jest.resetAllMocks());
beforeEach(() => {
  sessionStorage.clear();
});

const adminInfo = {
  name: "user",
  pass: "newpass",
  firstName: "Test",
  lastName: "User",
  email: "xample@example.com",
  roles: [{ id: 1, roleName: "ADMIN" }],
};

describe("create client as admin", () => {
   beforeEach(async () => {
     sessionStorage.setItem("userInfo", JSON.stringify(adminInfo));
   });

   it("renders", () => {
      render(<CreateClient />);
   });

   it("matches snapshot DOM node structure", () => {
     const comp = render(<CreateClient />);

     expect(comp.asFragment()).toMatchSnapshot();
   });

   it("contains correct elements", () => {
      const comp = render(<CreateClient/>);

      const heading = comp.getByRole("heading");
      const inputs = comp.getAllByRole("textbox");
      const buttons = comp.getAllByRole("button");

      expect(heading).toHaveTextContent("Create Client");
      expect(inputs).toHaveLength(8);
      expect(comp.getByRole("textbox", {name: "Client Name *"})).toHaveValue("");
      expect(comp.getByRole("textbox", {name: "Phone #"})).toHaveValue("");
      expect(comp.getByRole("textbox", {name: "Address Line 1"})).toHaveValue("");
      expect(comp.getByRole("textbox", {name: "Address Line 2"})).toHaveValue("");
      expect(comp.getByRole("textbox", {name: "City"})).toHaveValue("");
      expect(comp.getByRole("textbox", {name: "State / Province"})).toHaveValue("");
      expect(comp.getByRole("textbox", {name: "Country"})).toHaveValue("");
      expect(comp.getByRole("textbox", {name: "ZIP Code"})).toHaveValue("");
      expect(buttons).toHaveLength(2);
      expect(buttons[0]).toHaveValue("Submit");
      expect(buttons[1]).toHaveValue("Reset");
   });

   it("reset btn clears form", () => {
      const comp = render(<CreateClient />);

      const clientName = comp.getByRole("textbox", {name: "Client Name *"});
      const phone = comp.getByRole("textbox", {name: "Phone #"});
      const add1 = comp.getByRole("textbox", {name: "Address Line 1"});
      // const add2 = comp.getByRole("textbox", {name: "Address Line 2"});
      // const city = comp.getByRole("textbox", {name: "City"});
      // const state = comp.getByRole("textbox", {name: "State / Province"});
      // const country = comp.getByRole("textbox", {name: "Country"});
      // const zip = comp.getByRole("textbox", {name: "ZIP Code"});
      const reset = comp.getByRole("button", {name: "Reset"});

      userEvent.type(clientName, "Gap");
      // userEvent.type(phone, 1234567890);
      userEvent.type(add1, "A Street");

      expect(clientName).toHaveValue("Gap");
      // expect(phone).toHaveValue("+1(123) 456-7890");
      expect(add1).toHaveValue("A Street");

      userEvent.click(reset);
      expect(clientName).toHaveValue("");
      // expect(phone).toHaveValue("");
      expect(add1).toHaveValue("");
      
   })
})