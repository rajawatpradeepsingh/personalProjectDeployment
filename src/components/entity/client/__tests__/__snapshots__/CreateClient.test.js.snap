// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create client as admin matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    id="comp-content"
  >
    <div
      class="comp-header"
    >
      <h1>
        Create Client
      </h1>
    </div>
    <div
      class="create-client-error"
    >
      <div
        class="error-message"
      />
    </div>
    <div>
      <form
        enctype="multipart/form-data"
        id="create-client-form"
      >
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="clientName"
          >
            Client Name 
            <span
              class="requiredsign"
            >
              *
            </span>
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="clientName"
            maxlength="20"
            name="clientName"
            pattern="[^0-9]+"
            required=""
            title="Input can only contain letters"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="phoneNumber"
          >
            Phone # 
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="phoneNumber"
            maxlength="16"
            name="phoneNumber"
            title="Input can only accept numbers"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="addressLine1"
          >
            Address Line 1 
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="addressLine1"
            maxlength="200"
            name="addressLine1"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="addressLine2"
          >
            Address Line 2 
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="addressLine2"
            maxlength="200"
            name="addressLine2"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="city"
          >
            City 
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="city"
            maxlength="40"
            name="city"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="state"
          >
            State / Province 
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="state"
            name="state"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="country"
          >
            Country 
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="country"
            name="country"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="postalCode"
          >
            ZIP Code 
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="postalCode"
            name="postalCode"
            type="text"
            value=""
          />
        </div>
        <div
          class="create-int-btn-row"
        >
          <input
            class="form-btn"
            data-testid="submit-btn"
            id="int-save-btn"
            name="submit"
            type="submit"
            value="Submit"
          />
          <input
            class="form-btn"
            data-testid="reset-btn"
            id="int-reset-btn"
            name="cancel"
            type="reset"
            value="Reset"
          />
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
